src/grand-search-daemon/searcher/web/statictextworker.cpp

```diff
@@ -125,14 +125,19 @@ QString StaticTextWorker::createUrl(const QString &searchEngine) const
     static const QString baidu("https://www.baidu.com/s?&wd=%0");
     static const QString google("https://www.google.com/search?q=%0");
 
+    auto config = Configer::instance()->group(GRANDSEARCH_WEB_GROUP);
+    QString searchEngineAddr = config->value(GRANDSEARCH_WEB_SEARCHENGINE_CUSTOM_ADDR, QString(""));
```

Comment:
It needs a default url if the config value is invalid.

Reply:
1. the config value is blank, we can return `baidu` or `google`.
2. the config value is not a url, how can I handle it?
3. the config value is not include `%0`, how can I handle it?

and the lineedit needs to check the format

Reply:
and need a placeholder to prompt the user replace the keyword with `%0`

Reply:
yes, return baidu or google in value is blank.
And the user who write custom url for web engine will ensure that the url contain %0 to let grand search repalce key words.